src/webserver/http.rs

```diff
@@ -450,56 +497,15 @@ pub async fn main_handler(
         let if_modified_since = IfModifiedSince::parse(&service_request).ok();
 
         // Try to serve `path` as file, fall back to `404.sql` handler if it exists
-        let response = match serve_file(&path, &app_state, if_modified_since).await {
+        return match serve_file(&path, &app_state, if_modified_since).await {
             // Everything fine, static file can be served!
-            Ok(response) => response,
+            Ok(response) => Ok(service_request.into_response(response)),
 
             // File could not be served. Try to find a user provide 404 handler in the form of a
             // `404.sql` in the current directory. If there is none, look in the parent directeory,
             // and its parent directory, ...
-            Err(e) => {
-                let catch_all = "404.sql";
-                log::debug!("Trying to find a {catch_all:?} handler for {path:?}");
-                let mut full_path = path.into_owned();
-
-                loop {
-                    if full_path.is_empty() {
-                        log::debug!(
-                            "There is no {catch_all:?} handler, this response is terminally failed"
-                        );
-                        return Err(e);
-                    }
-
-                    let mut idx = 0;
-                    if let Some(idx_) = full_path.rfind('/') {
-                        full_path.truncate(idx_ + 1);
-                        idx = idx_;
-                    } else {
-                        full_path.clear();
-                    }
-
-                    full_path.push_str(&catch_all);
-
-                    let sql_path = PathBuf::from(&full_path);
-
-                    let app_state: &web::Data<AppState> =
-                        service_request.app_data().expect("app_state");
-                    if let Err(_) = app_state
-                        .sql_file_cache
-                        .get_with_privilege(app_state, &sql_path, false)
-                        .await
-                    {
-                        log::trace!("The 404 handler {sql_path:?} does not exist");
-                        full_path.truncate(idx);
-                        continue;
-                    };
-
-                    log::debug!("Processing SQL request via fallback: {:?}", sql_path);
-                    return process_sql_request(service_request, sql_path).await;
-                }
-            }
+            Err(e) => serve_fallback(service_request, e).await,
```

Comment:
I think we should call the 404 handler only if the original error is a 404; not for any error

Reply:
Done.